src/main/java/uk/gov/companieshouse/controller/DissolutionController.java

```diff
@@ -48,17 +64,24 @@ public DissolutionCreateResponse submitDissolutionRequest(
             @Valid @RequestBody final DissolutionCreateRequest body,
             HttpServletRequest request) {
 
-        if (StringUtils.isBlank(userId)) {
-            throw new UnauthorisedException();
-        }
+        logger.debug("[POST] Submitting dissolution request for company number {}", companyNumber);
 
-        if (dissolutionService.doesDissolutionRequestExistForCompany(companyNumber)) {
-            throw new ConflictException();
-        }
+        try {
+            if (StringUtils.isBlank(userId)) {
+                throw new MissingEricHeadersException();
```

Comment:
why are you throwing these new exceptions here? Why not throw the UnauthorisedException directly? Seems like additional complexity with no real benefit. The controller should only need to catch exceptions thrown from the dissolution service.